### DISCUSSION TITLE
HELP : MainWindow content not displaying , Avalonia 11 + macOS (m1)

### DISCUSSION THREAD

**Discussion by liam31012** [AvaloniaUI/Avalonia#16600] [Q&A]
I am using `Avaloina(11.1.1)` on macOS(m1)
here is my code 
**references**
```
        <PackageReference Include="Avalonia" Version="11.1.1" />
        <PackageReference Include="Avalonia.Desktop" Version="11.1.1"/>
        <PackageReference Include="Avalonia.Diagnostics" Version="11.1.1" />
        <PackageReference Include="Avalonia.Fonts.Inter" Version="11.1.1" />
        <PackageReference Include="Avalonia.ReactiveUI" Version="11.1.1"/>
        <PackageReference Include="Avalonia.Skia" Version="11.1.1" />
        <PackageReference Include="Avalonia.Themes.Fluent" Version="11.1.1" />
        <PackageReference Include="Avalonia.Themes.Simple" Version="11.1.1" />
```
**code**
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:PortApp.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="PortApp.Views.MainWindow"
        Icon="/Assets/avalonia-logo.ico"
        Title="PortApp">

    <Design.DataContext>
        <vm:MainWindowViewModel />
    </Design.DataContext>
    
    <TextBlock Text="TB 1" FontSize="25"/>
    
</Window>
```

the main window :
<img width="795" alt="截屏2024-08-05 21 30 45" src="https://github.com/user-attachments/assets/ce56d8cf-465a-4de2-8d9c-7af20b8c35e9">

why the `TextBlock` not displaying ? how to fix it

**Comment by thevortexcloud** [comment]
Have you added a theme to your app XAML? EG

https://docs.avaloniaui.net/docs/basics/user-interface/styling/themes/fluent

**Reply by liam31012** [reply]
Yes, I have , here is my code :
```
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:local="using:PortApp"
             x:Class="PortApp.App" RequestedThemeVariant="Light" >
    <Application.DataTemplates>
        <local:ViewLocator/>
    </Application.DataTemplates>
    
    
    <Application.Styles>
        <SimpleTheme />
    </Application.Styles>
    
</Application>
```

**Reply by thevortexcloud** [reply]
Does the text block even show up in the window if you open dev tools (press F12 to open them)?

**Reply by liam31012** [reply]
no dev tools after pressing F12, I got stuck on this.

**Reply by thevortexcloud** [reply]
You need the diagnostics package installed and for the `AttachDevTools` method to be called in your view's code behind. It _normally_ gets generated automatically, but you can add it manually

https://docs.avaloniaui.net/docs/guides/implementation-guides/developer-tools.

**Reply by liam31012** [reply]
I installed the nuget pkg `<PackageReference Include="Avalonia.Diagnostics" Version="11.1.1" />`, and the code about `AttachDevtool`that gets generated,

**Reply by thevortexcloud** [reply]
If it's still not opening when pressing F12, my guess is something is not initialising properly. Have you tested if a fresh project works? Run the following commands to generate a project with the latest templates:

`dotnet new install Avalonia.Templates`
`dotnet new update`
`dotnet new avalonia.app`

**Reply by liam31012** [reply]
haha bro, it works , `dotnet new install Avalonia.Templates
dotnet new update
dotnet new avalonia.app`
thx ,love u

**Reply by thevortexcloud** [reply]
You are welcome! 

FYI if you want an MVVM project instead, run `dotnet new avalonia.mvvm`
